engine/table/src/test/java/io/deephaven/engine/table/impl/updateby/TestEmStd.java

```diff
@@ -0,0 +1,1277 @@
+package io.deephaven.engine.table.impl.updateby;
+
+import io.deephaven.api.updateby.BadDataBehavior;
+import io.deephaven.api.updateby.OperationControl;
+import io.deephaven.api.updateby.UpdateByControl;
+import io.deephaven.api.updateby.UpdateByOperation;
+import io.deephaven.chunk.Chunk;
+import io.deephaven.chunk.ObjectChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.rowset.RowSetFactory;
+import io.deephaven.engine.table.PartitionedTable;
+import io.deephaven.engine.table.Table;
+import io.deephaven.engine.table.impl.QueryTable;
+import io.deephaven.engine.table.impl.TableDefaults;
+import io.deephaven.engine.table.impl.locations.TableDataException;
+import io.deephaven.engine.table.impl.util.ColumnHolder;
+import io.deephaven.engine.testutil.EvalNugget;
+import io.deephaven.engine.testutil.TstUtils;
+import io.deephaven.engine.testutil.generator.CharGenerator;
+import io.deephaven.engine.testutil.generator.SortedDateTimeGenerator;
+import io.deephaven.engine.testutil.generator.TestDataGenerator;
+import io.deephaven.engine.updategraph.UpdateGraphProcessor;
+import io.deephaven.engine.util.TableDiff;
+import io.deephaven.test.types.OutOfBandTest;
+import io.deephaven.time.DateTime;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.MathContext;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Random;
+
+import static io.deephaven.engine.testutil.GenerateTableUpdates.generateAppends;
+import static io.deephaven.engine.testutil.TstUtils.*;
+import static io.deephaven.engine.testutil.testcase.RefreshingTableTestCase.simulateShiftAwareStep;
+import static io.deephaven.engine.util.TableTools.*;
+import static io.deephaven.function.Basic.isNull;
+import static io.deephaven.time.DateTimeUtils.MINUTE;
+import static io.deephaven.time.DateTimeUtils.convertDateTime;
+import static io.deephaven.util.QueryConstants.*;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@Category(OutOfBandTest.class)
+public class TestEmStd extends BaseUpdateByTest {
```

Comment:
Can you add a test that hits the `rowRedirection != null` code paths?

Reply:
See above, can't actually be reached when `rowRedirection != null`.  Currently have `testZeroKeyAppendOnlyRedirected()`, `testZeroKeyGeneralRedirected()`, `testBucketedAppendOnlyRedirected()`, `testBucketedGeneralRedirected()` that test the redirected code paths.